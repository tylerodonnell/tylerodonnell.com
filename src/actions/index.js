import setLayout from './set-layout'
import * as TYPES from './types'

export {
  setLayout,
  TYPES,
}
