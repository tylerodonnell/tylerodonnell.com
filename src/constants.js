export const BREAKPOINTS = {
  SMALL: '576px',
  MEDIUM: '768px',
  LARGE: '992px',
}

export const COLORS = {
  BLUE: '#3B82F6',
  GRAY: '#1F2937',
  WHITE: '#FFFFFF',
}

export const FONT_FAMILIES = {
  MONOSPACE: 'Roboto Mono, monospace',
}

export const FONT_SIZES = {
  MEDIUM: '1rem',
  LARGE: '1.5rem',
  XLARGE: '2.4rem',
}

export const FONT_WEIGHTS = {
  LIGHT: 300,
}

export const PARTICLE_PER_SQUARE_PIXELS = 25000

export const TIMINGS = {
  SHORT: '200ms',
  MEDIUM: '500ms',
  LONG: '1000ms',
}

export const ZINDEX = {
  BACKGROUND: 1,
  CONTENT: 2,
}
