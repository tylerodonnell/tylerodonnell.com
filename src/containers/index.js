import Hero from './hero'
import Links from './links'
import Particles from './particles'

export {
  Hero,
  Links,
  Particles
}
