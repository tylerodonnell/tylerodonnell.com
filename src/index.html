<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    />
    <meta
      name="theme-color"
      content="#2B2D36"
    />
    <meta
      name="description"
      content="8+ years of experience as a strategic security leader with a developer background and expertise in product security, security education, and security operations"
    />
    <title>Tyler O'Donnell</title>
    <link
      href="https://fonts.gstatic.com"
      rel="preconnect"
    >
    <link
      href="https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@300;600&display=swap"
      rel="stylesheet"
    >
    <style>
      html {
        background: #282A36;
      }
    </style>
  </head>
  <body>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id="root"></div>
  </body>
</html>
