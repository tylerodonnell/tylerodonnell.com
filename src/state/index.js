import { always, compose } from 'ramda'
import React, { createContext, useContext, useReducer } from 'react'

import reducer from './reducer'
import initialState from './initialState'

export const StateContext = createContext()

export const StateProvider = ({ children }) => (
  <StateContext.Provider value={useReducer(reducer, initialState)}>
    {children}
  </StateContext.Provider>
)

export const withState = compose(useContext, always(StateContext))
