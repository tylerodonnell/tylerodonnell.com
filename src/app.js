import React from 'react'
import { StateProvider } from './state'
import Layout from './layout'

export default () => (
  <StateProvider>
    <Layout />
  </StateProvider>
)
