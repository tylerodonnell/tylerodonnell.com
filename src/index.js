import React from 'react'
import ReactDom from 'react-dom'

import App from './app'
import { StateProvider } from './state'

ReactDom.render(
  <React.StrictMode>
    <StateProvider>
      <App />
    </StateProvider>
  </React.StrictMode>,
  document.getElementById('root')
)
