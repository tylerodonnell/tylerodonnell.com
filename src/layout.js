import { compose } from 'ramda'
import React, { useEffect } from 'react'
import styled, { createGlobalStyle } from 'styled-components'

import { setLayout } from './actions'
import { BREAKPOINTS, COLORS, TIMINGS } from './constants'
import {
  Hero,
  Links,
  Particles
} from './containers'
import { layout as layoutUtil } from './utils'
import { withState } from './state'

const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }

  html {
    background: ${COLORS.GRAY};
    font-size: 13px;

    @media (min-width: ${BREAKPOINTS.SMALL}) {
      font-size: 14px;
    }

    @media (min-width: ${BREAKPOINTS.MEDIUM}) {
      font-size: 15px;
    }

    @media (min-width: ${BREAKPOINTS.LARGE}) {
      font-size: 16px;
    }
  }

  a {
    text-decoration: none;
    color: ${COLORS.WHITE};
    transition: color ${TIMINGS.SHORT} ease-in-out;
  }

  a:hover {
    color: ${COLORS.BLUE}
  }
`

const Layout = styled.main`
  align-items: center;
  display: flex;
  flex-direction: column;
  flex: 1;
  height: ${({ height }) => height}px;
  justify-content: center;
  max-height: 100vh;
  overflow: hidden;
  padding: 15px;
  position: relative;
  width: 100vw;
`

export default () => {
  const { dispatch, layout } = withState()

  useEffect(() => window.addEventListener('resize', compose(
    dispatch,
    setLayout,
    layoutUtil.calc
  )), [])

  return (
    <>
      <GlobalStyle />
      <Layout height={layout.height}>
        <Hero />
        <Links />
        <Particles />
      </Layout>
    </>
  )
}
