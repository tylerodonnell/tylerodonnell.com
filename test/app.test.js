/* global describe, expect, it */
import React from 'react'
import { render } from '@testing-library/react'

import App from '../src/app'
import { StateProvider } from '../src/state'

describe('<App />', () => {
  it('renders', () => {
    const app = render(
      <StateProvider>
        <App />
      </StateProvider>
    )

    expect(app.baseElement).toMatchSnapshot()
  })
})
