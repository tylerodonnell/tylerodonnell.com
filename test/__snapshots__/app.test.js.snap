// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<body>
  <div>
    <main
      class="app__Layout-sc-15mc0h9-0 bZDnHb"
      height="768"
    >
      <section
        class="hero__Hero-sc-178f7k9-2 dQUQSr"
      >
        <h1
          class="hero__Headline-sc-178f7k9-0 dUBxcK"
        >
          Tyler O'Donnell
        </h1>
        <h2
          class="hero__Title-sc-178f7k9-1 fZnhZT"
        >
          Security Engineering Manager at 
          <a
            class="hero__Role-sc-178f7k9-3 bbaiGI"
            href="https://articulate.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            Articulate
          </a>
        </h2>
      </section>
      <section
        class="links__Links-sc-1pi2t9e-0 ibZnjr"
      >
        <a
          class="link__Link-sc-1af9yan-0 fACVgQ"
          href="mailto:tyler@tylerodonnell.com"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
            />
          </svg>
        </a>
        <a
          class="link__Link-sc-1af9yan-0 fACVgQ"
          href="https://github.com/tylerodonnell"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </a>
        <a
          class="link__Link-sc-1af9yan-0 fACVgQ"
          href="https://www.linkedin.com/in/tylerodonnell"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            />
          </svg>
        </a>
        <a
          class="link__Link-sc-1af9yan-0 fACVgQ"
          href="https://keybase.io/tylerodonnell"
          target="_blank"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M286.17 419a18 18 0 1 0 18 18 18 18 0 0 0-18-18zm111.92-147.6c-9.5-14.62-39.37-52.45-87.26-73.71q-9.1-4.06-18.38-7.27a78.43 78.43 0 0 0-47.88-104.13c-12.41-4.1-23.33-6-32.41-5.77-.6-2-1.89-11 9.4-35L198.66 32l-5.48 7.56c-8.69 12.06-16.92 23.55-24.34 34.89a51 51 0 0 0-8.29-1.25c-41.53-2.45-39-2.33-41.06-2.33-50.61 0-50.75 52.12-50.75 45.88l-2.36 36.68c-1.61 27 19.75 50.21 47.63 51.85l8.93.54a214 214 0 0 0-46.29 35.54C14 304.66 14 374 14 429.77v33.64l23.32-29.8a148.6 148.6 0 0 0 14.56 37.56c5.78 10.13 14.87 9.45 19.64 7.33 4.21-1.87 10-6.92 3.75-20.11a178.29 178.29 0 0 1-15.76-53.13l46.82-59.83-24.66 74.11c58.23-42.4 157.38-61.76 236.25-38.59 34.2 10.05 67.45.69 84.74-23.84.72-1 1.2-2.16 1.85-3.22a156.09 156.09 0 0 1 2.8 28.43c0 23.3-3.69 52.93-14.88 81.64-2.52 6.46 1.76 14.5 8.6 15.74 7.42 1.57 15.33-3.1 18.37-11.15C429 443 434 414 434 382.32c0-38.58-13-77.46-35.91-110.92zM142.37 128.58l-15.7-.93-1.39 21.79 13.13.78a93 93 0 0 0 .32 19.57l-22.38-1.34a12.28 12.28 0 0 1-11.76-12.79L107 119c1-12.17 13.87-11.27 13.26-11.32l29.11 1.73a144.35 144.35 0 0 0-7 19.17zm148.42 172.18a10.51 10.51 0 0 1-14.35-1.39l-9.68-11.49-34.42 27a8.09 8.09 0 0 1-11.13-1.08l-15.78-18.64a7.38 7.38 0 0 1 1.34-10.34l34.57-27.18-14.14-16.74-17.09 13.45a7.75 7.75 0 0 1-10.59-1s-3.72-4.42-3.8-4.53a7.38 7.38 0 0 1 1.37-10.34L214 225.19s-18.51-22-18.6-22.14a9.56 9.56 0 0 1 1.74-13.42 10.38 10.38 0 0 1 14.3 1.37l81.09 96.32a9.58 9.58 0 0 1-1.74 13.44zM187.44 419a18 18 0 1 0 18 18 18 18 0 0 0-18-18z"
            />
          </svg>
        </a>
      </section>
      <div
        class="particles__StyledParticles-sc-11ot91b-0 kdTmAi"
        id="tsparticles"
      >
        <canvas
          class="tsparticles-canvas-el"
          height="0"
          style="width: 1024px; height: 768px;"
          width="0"
        />
      </div>
    </main>
  </div>
</body>
`;
